altair/vegalite/v5/api.py

```diff
@@ -56,7 +56,7 @@ def _dataset_name(values: Union[dict, list, core.InlineDataset]) -> str:
         values = values.to_dict()
     if values == [{}]:
         return "empty"
-    values_json = json.dumps(values, sort_keys=True)
+    values_json = json.dumps(values, sort_keys=True, default=str)
```

Comment:
For computing the values hash, I think it's fine to fallback to string representation for types not supported by `json.dumps` (`datetime.date` in this case).